src/infra/core/src/use_cases/view_and_edit_dataset_use_cases_impl.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use dill::{component, interface};
+use internal_error::{ErrorIntoInternal, InternalError};
+use kamu_core::auth::{DatasetAction, DatasetActionAuthorizer, DatasetActionUnauthorizedError};
+use kamu_core::{
+    DatasetRegistry,
+    EditDatasetUseCase,
+    EditDatasetUseCaseError,
+    EditMultiResult,
+    ViewDatasetUseCase,
+    ViewDatasetUseCaseError,
+    ViewMultiResult,
+};
+use odf::dataset::GetDatasetError;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[component(pub)]
+#[interface(dyn ViewDatasetUseCase)]
+#[interface(dyn EditDatasetUseCase)]
+pub struct ViewAndEditDatasetUseCasesImpl {
+    dataset_registry: Arc<dyn DatasetRegistry>,
+    dataset_action_authorizer: Arc<dyn DatasetActionAuthorizer>,
+}
```

Comment:
It seems having a shared helper between 2 use cases is a more natural form of decomposition.
I.e., a reason not to do common `execute` is an error type - it can't be identical, even if it consist of similar building blocks (access, internal).

Reply:
> I.e., a reason not to do common execute is an error type - it can't be identical, even if it consist of similar building blocks (access, internal).

Yes, that's right -- thanks for pointing that out

---

Shared helper (`AccessDatasetHelper`) has been extracted 

Done